pandas/core/ops.py

```diff
@@ -439,6 +461,18 @@ def f(x):
         self.lvalues = lvalues
         self.rvalues = rvalues
 
+
+    def _is_offset(self, arr_or_obj):
+        """ check if obj or first element of list-like is DateOffset """
+        if isinstance(arr_or_obj, pd.DateOffset):
+            return True
+        elif isinstance(arr_or_obj, pd.Series):
+            return isinstance(arr_or_obj.values[0], pd.DateOffset)
+        elif is_list_like(arr_or_obj):
```

Comment:
should check that all elements are offsets. there is an edge case where u have an array of date offsets (which may or may not be the same); I think in these cases u cannot use the vectorizes tricks
so what u are testing for here is a single offset